ProductsDatabase.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System.Data.SqlClient;
+
+public class ProductsDatabase
+{
+    private static readonly Lazy<ProductsDatabase> _lazyInstance = new Lazy<ProductsDatabase>(() => new ProductsDatabase());
+    private static SqlConnection sqlConnection;
+    private string connectionString = "Data Source = MOHAMMAD; " +
+        "Initial Catalog = Inventory; " +
+        "Integrated Security = True;";
+
+    public static ProductsDatabase Instance => _lazyInstance.Value;
+
+    private ProductsDatabase()
+    {
+        sqlConnection = new SqlConnection(connectionString);
+        sqlConnection.Open();
+        CreateProductsTable();
```

Comment:
Your code opens a single connection when the ProductsDatabase is created and keeps it open.
it is not a best practice to keep connection open.
here u need to close the connection and open it for each transaction you need to perform.